trieste/acquisition/sampler.py

```diff
@@ -276,3 +279,190 @@ def sample(self, at: TensorType, *, jitter: float = DEFAULTS.JITTER) -> TensorTy
+    posterior distribution for :math:`\theta` following their Bayesian linear regression motivation,
+    i.e. the computation revolves around an O(m^3)  inversion of a design matrix.
+
+    If :math:`n<m` then we use the kernel trick to recast computation to revolve around an O(n^3)
+    inversion of a gram matrix. As well as being more efficient in early BO
+    steps (where :math:`n<m`), this second computation method allows must larger choices
+    of m (as required to approximate very flexible kernels).
+
+    """
+
+    def __init__(self, dataset: Dataset, model: GPflowPredictor, num_features: int = 1000):
+        """
+        :param dataset: The data from the observer. Must be populated.
+        :param model: The model to sample from.
+        :param num_features: The number of features used to approximate the kernel.
+        :raise ValueError: If ``dataset`` is empty.
+        """
+        super().__init__(dataset, model)
+        self._num_features = num_features  # [m]
+        self._num_data = len(self._dataset.query_points)  # [n]
+
+        try:
+            self._noise_variance = model.get_observation_noise()
+            self._kernel = model.get_kernel()
+        except NotImplementedError:
+            raise ValueError(
+                """
+            Thompson sampling with random Fourier features only currently supports models
+            with a likelihood.variance attribute and an accessible kernel.
+            """
+            )
+
+        self._feature_functions = RandomFourierFeatures(
+            self._kernel, self._num_features, dtype=self._dataset.query_points.dtype
+        )
+
+        if (
+            self._num_features < self._num_data
+        ):  # if m < n  then calculate posterior in gram space through an m*m matrix inversion
+            self._theta_posterior = self._prepare_theta_posterior_in_design_space()
+        else:  # if n < m  then calculate posterior in gram space through an m*m matrix inversion
+            self._theta_posterior = self._prepare_theta_posterior_in_gram_space()
+
+    def _prepare_theta_posterior_in_design_space(self) -> tfp.distributions.Distribution:
+        r"""
+        Calculate the posterior of theta (the feature weights) in the design space. This
+        distribution is a Gaussian
+
+        .. math:: \theta \sim N(D^{-1}\Phi^Ty,D^{-1}\sigma^2)
+
+        where the [m,m] design matrix :math:`D=(\Phi^T\Phi + \sigma^2I_m)` is defined for
+        the [n,m] matrix of feature evaluations across the training data :math:`\Phi`
+        and observation noise variance :math:`\sigma^2`.
+
+        :return: The posterior distribution for theta.
+        """
+
+        phi = self._feature_functions(self._dataset.query_points)  # [n, m]
+        design_matrix = tf.matmul(phi, phi, transpose_a=True)  # [m, m]
+        s = self._noise_variance * tf.eye(self._num_features, dtype=phi.dtype)
+        inv_design_matrix = tf.linalg.inv(design_matrix + s)
```

Comment:
I think it would be better to avoid inverting the matrix. If you get the cholesky factorisation of the design matrix you can use the `cholesky_solve` function to compute the mean and covariance.

Reply:
Thanks, this is the sort of wisdom I wanted!